src/gui/styles/dark/DarkStyle.cpp

```diff
@@ -104,18 +118,11 @@ void DarkStyle::polish(QWidget* widget)
     if (qobject_cast<QMainWindow*>(widget) || qobject_cast<QDialog*>(widget) || qobject_cast<QMenuBar*>(widget)
         || qobject_cast<QToolBar*>(widget)) {
         auto palette = widget->palette();
-#if defined(Q_OS_MACOS)
-        if (osUtils->isDarkMode()) {
-            // Let the Cocoa platform plugin draw its own background
-            palette.setColor(QPalette::All, QPalette::Window, Qt::transparent);
```

Comment:
Removing this breaks native theming. You can see that in your screenshot. The gradient is gone and there is a line between window title bar and chrome body.

Reply:
Then we can only do transparent on Catalina and above

Reply:
The way we have it now is that it's transparent if the theme matches and not transparent otherwise. That should take care of older macOS versions, too. The native platform theme does the same trick to let the window chrome shine through. Perhaps they are doing something in addition to fix the flyout background.

Reply:
This causes the extended toolbar to be transparent and older mac versions to show black background. In general the previous implementation was not good.